tools/run_tests/run_tests.py

```diff
@@ -477,7 +477,7 @@ def _compiler_options(self, use_docker, compiler):
             _check_compiler(compiler, ['default', 'cmake'])
 
         if compiler == 'default' or compiler == 'cmake':
-            return ('debian11', [])
+            return ('debian11', ["-DCMAKE_CXX_STANDARD=14"])
```

Comment:
It feels odd that to our "default" build via run_tests.py we need to pass extra options.
Would that mean that  if one builds with cmake by hand, without passing any extra configuration, they would get a differently configured build that if they run run_tests.py? If so, that feels wrong.

Reply:
This is required because of Apple clang which uses C++98 by default (yay). Also this is only for tests and gRPC library targets don't specify it so that C++ version can be picked by applications.